libs/ts-rest/core/src/lib/client.ts

```diff
@@ -319,6 +353,20 @@ export const getRouteQuery = <TAppRoute extends AppRoute>(
         ...headers,
       },
     });
+
+    if (!clientArgs.strict) {
+      return await request;
+    }
+    const response = await request;
+
+    if (knownResponseStatuses.includes(response.status.toString())) {
+      return response;
+    }
+    throw new Error(
```

Comment:
Would be helpful to throw a custom exception here exposing the response object through a property

Reply:
Agreed! Would you be open to having the exception configurable? I use an error library that captures additional information using async hooks and it would be helpful to be able to use one of these.

Reply:
I think that might be a too specific feature, that could otherwise be covered by the `api` handler option, where you can catch the exception and re-throw your custom error.

```typescript
const client = initQueryClient(postsApi, {
  ...
  api: async (args) => {
    try {
      return tsRestFetchApi(args);
    } catch (error) {
      throw CustomException(error);
    }
  },
});
```

Reply:
I wondered about that option but unless I am mistaken I don't think this would work as show above.

`getRouteQuery` is where this logic is implemented and it calls `fetchApi` which uses the custom api implementation.


Something like the following could work I think
```ts
const client = initQueryClient(postsApi, {
  ...
  api: async (args) => {
      const response = await tsRestFetchApi(args);
      if (!responseStatusExpected(response)) {
        throw CustomException("wrong status code")
      }
      return response
  },
});
```

But the hard part is knowing how to implement `responseStatusExpected` as I don't think the context is known at this point. The expected statuses could be passed down into the `api` function?

Reply:
@gingermusketeer Oh you're right! It could be a simple change though, just need pass the `AppRoute` object along with the rest of the args. I think I can include this as part of a PR I am currently working on.